src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -1,17 +1,121 @@
 package wolox.training.controllers;
 
-import org.springframework.stereotype.Controller;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.repositories.IBookRepository;
 
-@Controller
+
+/**
+ * This class is the service responsible of handling the requests related to the book api
+ */
+
+@RestController
+@RequestMapping("/api/books")
 public class BookController {
 
+    @Autowired
+    private IBookRepository bookRepository;
+
     @GetMapping("/greeting")
-    public String greeting(@RequestParam(name="name", required=false, defaultValue="Poli") String name, Model model) {
+    public String greeting(@RequestParam(name = "name", required = false, defaultValue = "Poli") String name,
+            Model model) {
         model.addAttribute("name", name);
         return "greeting";
     }
 
+    /**
+     * @return all the books contained in the model
+     */
+    @GetMapping("/all")
```

Comment:
When we want to get the all items from a resource, we should use the root path of the resource. See the best practices for [REST naming resources](https://docs.microsoft.com/en-us/azure/architecture/best-practices/api-design#organize-the-api-design-around-resources) to learn more about this.
```suggestion
    @GetMapping
```